src/inmanta/protocol/endpoints.py

```diff
@@ -191,11 +210,13 @@ async def start(self) -> None:
         """
         assert self._env_id is not None
         LOGGER.log(3, "Starting agent for %s", str(self.sessionid))
-        self._client = SessionClient(self.name, self.sessionid, timeout=self.server_timeout)
+        self._client = SessionClient(self.name, self.sessionid, timeout=self.server_timeout, force_instance=True)
+        self._heartbeat_client = SessionClient(self.name, self.sessionid, timeout=self.server_timeout, force_instance=True)
```

Comment:
I haven't looked too deep into this, but why is the force_instance parameter required ?

Reply:
if you don't use the force_instance paramter the session clients use the same pool. the force_instance forces the new session to be use a different pool than the existing one.

Reply:
Oh ok I completely missed that part, I'll have a deeper look with a fresh pair of eyes on monday :+1: